File: Difficulty: Easy/Move All Zeroes to End/move-all-zeroes-to-end.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
// User function template for C++
class Solution {
  public:
    void pushZerosToEnd(vector<int>& arr) {
        // code here
        vector<int>v;
        int c=0;
        for(int i=0;i<arr.size();i++)
        {
            if(arr[i]==0) c++;
            else v.push_back(arr[i]);
        }
        for(int i=0;i<c;i++)
        {
            v.push_back(0);
        }
        arr.clear();
        for(int i=0;i<v.size();i++)
        {
            arr.push_back(v[i]);
        }
    }
};

//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    cin.ignore();
    while (t--) {
        vector<int> arr;
        string input;
        getline(cin, input);
        stringstream ss(input);
        int number;
        while (ss >> number) {
            arr.push_back(number);
        }
        Solution ob;
        int n = arr.size();
        ob.pushZerosToEnd(arr);
        for (int i = 0; i < n; i++) {
            cout << arr[i] << " ";
        }
        cout << "\n";
    }
    return 0;
}
// } Driver Code Ends

File: Difficulty: Easy/Move All Zeroes to End/move-all-zeroes-to-end.java
class Solution {
    void pushZerosToEnd(int[] arr) {
        // code here
        int j=-1;
        for(int i=0;i<arr.length;i++){
            if(arr[i]==0){
                j=i;
                break;
            }
        }
        if(j!=-1){
            for(int i=j+1;i<arr.length;i++){
                if(arr[i]!=0){
                    int temp=arr[i];
                    arr[i]=arr[j];
                    arr[j]=temp;
                    j++;
                }
                else{
                    continue;
                }
            }
        }
    }
}